base_layer/mmr/src/common.rs

```diff
@@ -75,7 +75,15 @@ pub fn find_peaks(size: usize) -> Vec<usize> {
             sum_prev_peaks += peak_size;
             num_left -= peak_size;
         }
-        peak_size >>= 1;
+        // need to verify if other peaks exist on the same height, in which case
+        // the size doesn't generate a complete mmr, e.g.
+        //    2
+        //   / \
+        //  0   1   3   4
+        // where we have two peaks at height 0.
+        if num_left < peak_size {
```

Comment:
This is equivalent to an else branch on the previous if statement. I think the previous code was correct, though returning an vec! for an invalid size parameter should perhaps be an Option rather

Then in the merkle proof generation (`MerkleProof::generate_proof`), there was a missing check, you would do something like

```rust
if peaks.is_none() {
            return Err(MerkleProofError::Unexpected);
        }
```